### PR TITLE
TGP-2057: added controller for download trader data

### PR DESCRIPTION
**Pull request by william-eyre**

### CODE REVIEW

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/DownloadTraderDataController.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  service: DownloadTraderDataService,
+  override val controllerComponents: ControllerComponents,
+  override val uuidService: UuidService
+)(implicit val ec: ExecutionContext)
+    extends BackendBaseController
+    with ValidationRules {
+
+  def requestDataDownload(eori: String): Action[AnyContent] = authAction(eori).async {
+    implicit request: Request[AnyContent] =>
+      val result = for {
```
Comment (mauro-fanton) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/DownloadTraderDataController.scala:40]: We should validate the accept header here. The ticket has a bullet point about that.

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/service/DownloadTraderDataService.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+)(implicit ec: ExecutionContext)
+    extends Logging {
+
+  def requestDownload(eori: String)(implicit hc: HeaderCarrier): Future[Either[EisHttpErrorResponse, Int]] = {
+    val correlationId = uuidService.uuid
+    connector
+      .requestDownload(eori, correlationId)
+      .map {
+        case Right(_)       => Right(ACCEPTED)
+        case Left(response) => Left(response)
```
Comment (mauro-fanton) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/service/DownloadTraderDataService.scala:41]: we may want to add a logger.warn if the connector request return an error. I have found that when we investigate a bug in Kibana, there are no many logs.

File: it/test/uk/gov/hmrc/tradergoodsprofilesrouter/DownloadTraderDataIntegrationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        "code"          -> "FORBIDDEN",
+        "message"       -> "Forbidden"
+      )
+    }
+  }
+
+  private def stubForEis(httpStatus: Int) =
+    stubFor(
+      get(urlEqualTo(s"$pegaConnectorPath/$eori/download"))
+        .withHeader("X-Forwarded-Host", equalTo("MDTP"))
```
Comment (mauro-fanton) [inline it/test/uk/gov/hmrc/tradergoodsprofilesrouter/DownloadTraderDataIntegrationSpec.scala:81]: this is not testing the header, if this was the intent. If you remove all the .withHeader the test will pass anyway.

To test the header you may want to use the wireMock.verify with header function or you can do that in the Connector Spec.

Also I think there are few test case missing here, such as testing the error response json message for the different error.

File: test/uk/gov/hmrc/tradergoodsprofilesrouter/support/BaseConnectorSpec.scala [hunk 4]
```diff
@@ -121,7 +121,9 @@ trait BaseConnectorSpec extends PlaySpec with BeforeAndAfterEach with EitherValu
       getRecords = "/tgp/getrecords/v1",
       recordGetToken = "dummyRecordGetBearerToken",
       withdrawAdvise = "/tgp/withdrawaccreditation/v1",
-      withdrawAdviseToken = "dummyWithdrawAdviceBearerToken"
+      withdrawAdviseToken = "dummyWithdrawAdviceBearerToken",
+      downloadTraderData = "/tgp/downloadtraderdata/v1",
```
Comment (mauro-fanton) [inline test/uk/gov/hmrc/tradergoodsprofilesrouter/support/BaseConnectorSpec.scala:125]: is the URL for the downloadedTraderData right? the EPIDD have different url

File: conf/application.conf [hunk 5]
```diff
@@ -121,9 +121,11 @@ microservice {
         get-records  = "/tgp/getrecords/v1"
         request-advice  = "/tgp/createaccreditation/v1"
         withdraw-advice = "/tgp/withdrawaccreditation/v1"
+        download-trader-data = "/tgp/downloadtraderdata/v1" // TODO check this is right?
```
Comment (mauro-fanton) [inline conf/application.conf:124]: the EPIDD has the following URL: v1/tgp/record/{eori}/download

**Comment by mauro-fanton** [comment]
There are some error code missing such as 000, 001, 002, 003, 004, 005 as per EPIDD

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/DownloadTraderDataController.scala (hunk 1 above)
Comment (william-eyre) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/DownloadTraderDataController.scala:40]: There isn't an accept header in this request. It doesn't have a body or return anything

Comment (mauro-fanton) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/DownloadTraderDataController.scala:40]: oh, ok.

**Review by mauro-fanton — APPROVED** [review]

**Pull request merged**